tycho-its/src/test/java/org/eclipse/tycho/test/product/ProductBuildTest.java

```diff
@@ -9,17 +9,54 @@
  *******************************************************************************/
 package org.eclipse.tycho.test.product;
 
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
 import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 import org.apache.maven.it.Verifier;
+import org.eclipse.tycho.TychoConstants;
 import org.eclipse.tycho.test.AbstractTychoIntegrationTest;
 import org.junit.Test;
 
+import de.pdark.decentxml.Document;
+import de.pdark.decentxml.Element;
+import de.pdark.decentxml.XMLParser;
+
 public class ProductBuildTest extends AbstractTychoIntegrationTest {
 	@Test
 	public void testMavenDepedencyInTarget() throws Exception {
 		Verifier verifier = getVerifier("product.mavenLocation", false);
 		verifier.executeGoals(Arrays.asList("clean", "verify"));
 		verifier.verifyErrorFreeLog();
 	}
+
+	@Test
+	public void testPGPSignedInProduct() throws Exception {
+		Verifier verifier = getVerifier("product.pgp", false);
+		verifier.executeGoals(Arrays.asList("clean", "install"));
+		verifier.verifyErrorFreeLog();
+		checkPGP(verifier, "target/repository/artifacts.xml");
+		checkPGP(verifier, "target/products/pgp/linux/gtk/x86_64/artifacts.xml");
+
+	}
+
+	protected void checkPGP(Verifier verifier, String repositoryArtifacts) throws IOException {
+		File artifactXml = new File(verifier.getBasedir(), repositoryArtifacts);
+		assertTrue("required artifacts file " + artifactXml.getAbsolutePath() + " not found!", artifactXml.isFile());
+		Document artifactsDocument = XMLParser.parse(artifactXml);
+		Optional<Element> optional = artifactsDocument.getChild("repository").getChild("artifacts")
+				.getChildren("artifact").stream()
+				.filter(element -> element.getAttributeValue("id").equals("org.mockito.mockito-core")).findAny();
+		assertTrue("artifact org.mockito.mockito-core not found", optional.isPresent());
+		Element element = optional.get();
+		Map<String, String> properties = element.getChild("properties").getChildren("property").stream()
+				.collect(Collectors.toMap(e -> e.getAttributeValue("name"), e -> e.getAttributeValue("value")));
+		assertTrue(properties.containsKey(TychoConstants.PROP_PGP_SIGNATURES)
```

Comment:
I don't know if you have a property for this p2 constant:
```
	public static final String PGP_SIGNER_KEYS_PROPERTY_NAME = "pgp.publicKeys"; //$NON-NLS-1$
```
Both the signature and the key should be present and the **pgp.publicKeys**'s value is what's displayed to the user, similar to the certificate or a jar-signed artifact.

Reply:
Done,